modules/core/src/test/java/org/apache/ignite/internal/IgniteClientReconnectContinuousProcessorTest.java

```diff
@@ -305,6 +312,34 @@ public void testCacheContinuousQueryReconnectNewServer() throws Exception {
         }
     }
 
+    @Test
+    public void testCacheContinuousQuerySpliteratorMultipleCalls() {
+        Ignite client = grid(serverCount());
+
+        assertTrue(client.cluster().localNode().isClient());
+
+        IgniteCache<Object, Object> clientCache = client.getOrCreateCache(new CacheConfiguration<>(DEFAULT_CACHE_NAME));
+
+        CacheEventListener lsnr = new CacheEventListener();
+
+        ContinuousQuery<Object, Object> qry = new ContinuousQuery<>();
+
+        qry.setInitialQuery(new ScanQuery<>(new IgniteBiPredicate<Object, Object>() {
+            @Override public boolean apply(Object key, Object val) {
+                return key != null;
+            }
+        }));
+        qry.setAutoUnsubscribe(true);
+
+        qry.setLocalListener(lsnr);
+
+        try(QueryCursor<?> cur = clientCache.query(qry)) {
+            cur.iterator();
+            cur.spliterator();
```

Comment:
Note that an instance of `java.util.Spliterators$IteratorSpliterator` is returned here, and only the usage of deprecated `SqlQuery` will cause returning of `QueryKeyValueSpliterator` instance, it meets your expectations?

Reply:
Yes, I am fine with that - as long as it doesn't throw exception "Iterator is already fetched or query was cancelled."